src/domain/in-app-notification-reader/dto/in.app.notification.callout.published.ts

```diff
@@ -0,0 +1,28 @@
+import { ObjectType } from '@nestjs/graphql';
+import {
+  InAppNotificationCalloutPublishedPayload,
+  NotificationEventType,
+  InAppNotificationCategory,
+} from '@alkemio/notifications-lib';
+import { ICallout } from '@domain/collaboration/callout';
+import { ISpace } from '@domain/space/space/space.interface';
+import { InAppNotification } from '../in.app.notification.interface';
+import { InAppNotificationState } from '@domain/in-app-notification/in.app.notification.state';
+
+@ObjectType('InAppNotificationCalloutPublished', {
+  implements: () => InAppNotification,
+})
+export abstract class InAppNotificationCalloutPublished
```

Comment:
I think you can clearly have a base abstract class e.g. `InAppNotificationBase` - you have `type`, `category`, `id`, `state`, `triggeredAt`... already overlapping.

Reply:
that was my initial approach - but to enforce consistency the base class has to implement a common interface.
The code would look like similar to
```typescript
@ObjectType('InAppNotificationCalloutPublished', {
  implements: () => InAppNotification,
})
export abstract class InAppNotificationCalloutPublished extends InAppNotification {
// ... rest of code
```
and `InAppNotification` already uses each of the concrete classes in the `resolveType` method of the `InterfaceType `decorator.
That creates a circular dependency - for `InAppNotificationCalloutPublished` to extend `InAppNotification`, `InAppNotification` has to be instantiated first, which requires all order concrete classes as dependencies to that class to be instantiated first.

**Happy to work this issue through together.**